src/main/java/core/basesyntax/utils/transaction/handlers/impl/SupplyOperation.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.utils.transaction.handlers.impl;
+
+import core.basesyntax.storage.impl.StorageImpl;
+import core.basesyntax.utils.transaction.FruitTransaction;
+import core.basesyntax.utils.transaction.handlers.OperationHandler;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void perform(FruitTransaction transaction, StorageImpl storage) {
```

Comment:
The method signature for `perform` should use the interface `Storage` as the type for the `storage` parameter instead of the concrete implementation `StorageImpl` to ensure that the code adheres to the Dependency Inversion Principle.